purb/skip.go

```diff
@@ -0,0 +1,208 @@
+package purb
+
+import (
+	"fmt"
+	"github.com/dedis/crypto/random"
+)
+
+// Pick a uint32 uniformly at random
+func randUint32() uint32 {
+	return random.Uint32(random.Stream)
+}
+
+// Pick a random height for a new skip-list node from a suitable distribution.
+func skipHeight() int {
+	height := 1
+	for v := randUint32() | (1 << 31); v&1 == 0; v >>= 1 {
+		height++
+	}
+	return height
+}
+
+type skipNode struct {
```

Comment:
It is possible, but most (all?) of the methods are private. So to use it we would have to modify the source code, which might be a better solution then copying it. Although possibly we should consider removing nego, since PURB and PURB TLS is the evolution of what the nego code did/was meant to do.